libs/@guardian/identity-auth/src/identityAuth.ts

```diff
@@ -94,23 +94,31 @@ export class IdentityAuth<
 	 */
 	public async isSignedInWithAuthState(): Promise<IdentityAuthState<AC, IC>> {
 		try {
-			// if the user has a GU_SO cookie, they have recently signed out, so we should clear their tokens
-			// the GU_SO cookie will be automatically cleared when the user signs back in
-			if (getCookie({ name: 'GU_SO', shouldMemoize: true })) {
-				this.tokenManager.clear();
-				return {
-					accessToken: undefined,
-					idToken: undefined,
-					isAuthenticated: false,
-				};
-			}
-
-			// if user tokens already exist, they are signed in
+			// This follows the flowchart from https://github.com/guardian/gateway/blob/main/docs/okta/web-apps-integration-guide.md#how-to-know-if-a-reader-is-signed-in to determine if the user is signed in
+			// first check if the user has valid access and id tokens
 			const authState = this.authStateManager.getAuthState();
+			// if they do we need to check if a user has recently signed out
 			if (authState.isAuthenticated) {
 				// validate the id token and access token to make sure auth state is still valid
 				await this.token.verifyTokens(authState.idToken, authState.accessToken);
 
+				// if a user has a GU_SO cookie, they might have recently signed out
+				const guSoCookie = parseInt(
```

Comment:
If the `parseInt` call fails, this gets caught all the way at the end - is that acceptable?

Reply:
Perfect!